assets/js/components/projects/ProjectSettings.jsx

```diff
@@ -0,0 +1,366 @@
+import React, { Component, PropTypes } from "react"
+import { connect } from "react-redux"
+import { bindActionCreators } from "redux"
+import { withRouter } from "react-router"
+import { orderedItems } from "../../reducers/collection"
+import { InputWithLabel } from "../ui"
+import ColourSchemeModal from "./ColourSchemeModal"
+import TimezoneAutocomplete from "../timezones/TimezoneAutocomplete"
+import * as actions from "../../actions/projects"
+import * as projectActions from "../../actions/project"
+import { updateProject, fetchProject, updateProjectArchived } from "../../api"
+import merge from "lodash/merge"
+import dateformat from "dateformat"
+import { translate } from "react-i18next"
+import { isProjectReadOnly } from "../../reducers/project"
+import { ArchiveIcon } from "../ui"
+
+class ProjectSettings extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      name: "",
+      timezone: "",
+      colourScheme: "default",
+      initialSuccessRate: 0,
+      eligibilityRate: 0,
+      responseRate: 0,
+      validRespondentRate: 0,
+      detailedRates: false,
+      archiveAction: "archive",
+    }
+  }
+
+  fetchProjectAndSetState() {
+    const { projectId, t } = this.props
+    fetchProject(projectId).then((response) => {
+      const project = response.entities.projects[response.result]
+      this.setState({
+        name: project.name,
+        timezone: project.timezone,
+        colourScheme: project.colourScheme,
+        initialSuccessRate: project.initialSuccessRate,
+        eligibilityRate: project.eligibilityRate,
+        responseRate: project.responseRate,
+        validRespondentRate: project.validRespondentRate,
+        detailedRates: project.eligibilityRate != null,
+        archiveAction: project.readOnly ? "unarchive" : "archive",
+        readOnly: !project.owner,
+      })
+      this.initialState = this.state
+      this.configureRates(this.state.detailedRates)
+      this.project = project
+    })
+  }
+
+  componentDidMount() {
+    const { projectId, t } = this.props
+    this.fetchProjectAndSetState()
+  }
+
+  changePageSize(pageSize) {
+    const { projectId } = this.props
+  }
+
+  toggleDetailedRates(toggle) {
+    const { projectId } = this.props
+    const { detailedRates } = this.state
+    this.setState({ detailedRates: !detailedRates })
+    this.configureRates(!detailedRates)
+  }
+
+  configureRates(detailedRates) {
+    if (detailedRates) {
+      document.getElementById("initialSuccessRate").getElementsByTagName("input")[0].disabled = true
+      document.getElementById("eligibilityRate").style.display = "block"
+      document.getElementById("responseRate").style.display = "block"
+      document.getElementById("validRespondentRate").style.display = "block"
```

Comment:
You should never have to query the DOM. React handles the props & state and re-renders as needed what needs to be re-rendered.

Use `this.state.detailedRates` in `render()` directly to selectively render the 3 inputs for detailed rates. For example extract the rendering of these into a method, then in the main `render()`:

```jsx
{this.state.detailedRates && this.renderDetailedRates()}
```

 You already nullify the state values below, React will detect that thanks to `this.setState` and re-render.